collections/ipnskeys.go

```diff
@@ -0,0 +1,84 @@
+package collections
+
+import (
+	"context"
+
+	"github.com/textileio/go-threads/core/thread"
+	"github.com/textileio/textile/v2/badgerdb"
+	"github.com/textileio/textile/v2/model"
+	"github.com/textileio/textile/v2/mongodb"
+)
+
+type IPNSKeys struct {
+	hub bool
+	m   mongodb.IPNSKeys
+	b   badgerdb.IPNSKeys
+}
+
+type IPNSKeysOptions func(*IPNSKeys)
+
+func WithMongoIPNSKeysOpts(m mongodb.IPNSKeys) IPNSKeysOptions {
+	return func(i *IPNSKeys) {
+		i.m = m
+	}
+}
+
+func WithBadgerIPNSKeysOpts(b badgerdb.IPNSKeys) IPNSKeysOptions {
+	return func(i *IPNSKeys) {
+		i.b = b
+	}
+}
+
+func NewIPNSKeys(_ context.Context, hub bool, opts ...IPNSKeysOptions) (*IPNSKeys, error) {
+	k := &IPNSKeys{
+		hub: hub,
+	}
+
+	for _, opt := range opts {
+		opt(k)
+	}
+
+	return k, nil
+}
+
+func (k *IPNSKeys) Create(ctx context.Context, name, cid string, threadID thread.ID) error {
+	if k.hub {
+		return k.m.Create(ctx, name, cid, threadID)
```

Comment:
@jsonsivar 

What do you think about creating even a small interface like:

```
type NotAGoodName interface {
    Create(ctx, name, cid, threadID)
    Get(ctx, name)
    GetByCid(ctx, cid)
    ListByThreadID(ctx, threadID)
    Delete(ctx, name)
}
```

And using it in the `type IPNSKeys struct`?

(with proper signatures, of course)